packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/index.tsx

```diff
@@ -41,17 +34,26 @@ class Checkout extends PureComponent<Props> {
       )
       this.props.simpleBuyActions.createSBOrder(
         undefined,
-        formValues?.method?.type as SBPaymentMethodType['type']
+        this.props.method.type as SBPaymentMethodType['type']
       )
-    } else if (formValues && formValues.method) {
-      switch (formValues.method.type) {
+    } else if (formValues && !this.props.method) {
+      const fiatCurrency = this.props.fiatCurrency || 'USD'
+      this.props.simpleBuyActions.setStep({
+        step: 'PAYMENT_METHODS',
+        fiatCurrency,
+        pair: this.props.pair
+      })
+    } else if (formValues && this.props.method) {
```

Comment:
Can probably be just `else if (this.props.method)`

Reply:
there is a TODO to figure out `formValues.method.id`